src/components/graph/menus/create-node-menu.js

```diff
@@ -193,26 +234,27 @@ const CreateNodeMenu = ({
             onClose={handleClose}
         >
             {Object.values(NODE_MENU_ITEMS).map((item) => {
+                const hasSubMenuItem = item.subMenuItems !== undefined;
                 return (
                     (activeNode?.type !== 'ROOT' || item.onRoot) && (
-                        <MenuItem
-                            className={classes.menuItem}
-                            onClick={item.action}
-                            key={item.id}
-                            disabled={item.disabled}
-                        >
-                            <ListItemText
-                                key={item.id}
-                                className={classes.listItemText}
-                                primary={
-                                    <Typography noWrap>
-                                        {intl.formatMessage({
-                                            id: item.id,
-                                        })}
-                                    </Typography>
-                                }
-                            />
-                        </MenuItem>
+                        <>
+                            {!hasSubMenuItem && <SubMenuItem item={item} />}
+
+                            {hasSubMenuItem && (
+                                <NestedMenuItem
+                                    label={intl.formatMessage({
+                                        id: item.id,
+                                    })}
+                                    parentMenuOpen={true}
+                                >
+                                    {Object.values(item.subMenuItems).map(
```

Comment:
This is not recursive. If we add `subMenuItems` in an item which is already a sub menu item then it won't work.